test/contexts_test.dart

```diff
@@ -0,0 +1,96 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:sentry/sentry.dart';
+import 'package:test/test.dart';
+
+void main() {
+  group(Contexts, () {
+    final testBootTime = DateTime.fromMicrosecondsSinceEpoch(0);
+    test('serializes to JSON', () {
+      final testDevice = Device(
+        name: 'testDevice',
+        family: 'testFamily',
+        model: 'testModel',
+        modelId: 'testModelId',
+        arch: 'testArch',
+        batteryLevel: 23,
+        orientation: Orientation.landscape,
+        manufacturer: 'testOEM',
+        brand: 'testBrand',
+        screenResolution: '123x345',
+        screenDensity: '99',
+        screenDpi: '100',
+        online: false,
+        charging: true,
+        lowMemory: false,
+        simulator: true,
+        memorySize: 1234567,
+        freeMemory: 12345,
+        usableMemory: 9876,
+        storageSize: 1234567,
+        freeStorage: 1234567,
+        externalStorageSize: 98765,
+        externalFreeStorage: 98765,
+        bootTime: testBootTime,
+        timezone: 'Australia/Melbourne',
+      );
+      final testOS = OperatingSystem(name: 'testOS');
+      final testRuntimes = [
+        Runtime(name: 'testRT1', version: '1.0'),
+        Runtime(name: 'testRT2', version: '2.3.1'),
+      ];
+      final testApp = App(version: '1.2.3');
+      final testBrowser = Browser(version: '12.3.4');
+
+      final contexts = Contexts(
```

Comment:
Let's put this in an `Event` instance, then call `event.toJson()` and `expect(eventJson['contexts'], ...);`. This will also test the logic in `Event.toJson()`.

Reply:
@yjbanov test improved, changelog updated and version bumped in latest commit.